ksqldb-engine/src/main/java/io/confluent/ksql/planner/plan/PreJoinProjectNode.java

```diff
@@ -73,12 +83,16 @@ public LogicalSchema getSchema() {
 
   @Override
   public Optional<KeyFormat> getPreferredKeyFormat() {
-    return joiningSource.getPreferredKeyFormat();
+    if (joiningSource.isPresent()) {
+      return joiningSource.get().getPreferredKeyFormat();
+    }
+
+    return Optional.of(getSourceKeyFormat());
   }
 
   @Override
   public void setKeyFormat(final KeyFormat format) {
-    joiningSource.setKeyFormat(format);
+    joiningSource.ifPresent(source -> source.setKeyFormat(format));
```

Comment:
Not sure if I understand this change? If `source` is not `JoiningNode` but a `DataSourceNode`, why can we make this call a no-op?

Reply:
Wondering if we should throw an exception for this case instead of a no-op? Or could it happen that `setKeyFormat` is called for this case? (Seems, it should never be called for this case?)

Reply:
It will be called for this case. `setKeyFormat()` is called on all JoiningNodes.